interpreter/convert.go

```diff
@@ -18,39 +22,44 @@ func convertToStarlark(value interface{}) (starlark.Value, error) {
 	case uint64:
 		return starlark.MakeUint64(v), nil
 	case float32:
+		return starlark.Float(v), nil
 	case float64:
 		return starlark.Float(v), nil
 	case string:
 		return starlark.String(v), nil
-	case []interface{}:
-		var elems []starlark.Value
-		for _, elem := range v {
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
-			elems = append(elems, val)
-		}
-		return starlark.NewList(elems), nil
-	case map[interface{}]interface{}:
-		dict := starlark.NewDict(len(v))
-
-		for index, elem := range v {
-			key, err := convertToStarlark(index)
-			if err != nil {
-				return nil, err
+	default:
+		reflectV := reflect.ValueOf(value)
+		switch reflectV.Kind() {
+		case reflect.Slice:
+			var elems []starlark.Value
+			for i := 0; i < reflectV.Len(); i++ {
+				val, err := convertToStarlark(reflectV.Index(i).Interface())
+				if err != nil {
+					return nil, err
+				}
+				elems = append(elems, val)
 			}
+			return starlark.NewList(elems), nil
+		case reflect.Map:
+			dict := starlark.NewDict(len(reflectV.MapKeys()))
 
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
+			iter := reflectV.MapRange()
+			for iter.Next() {
+				key, err := convertToStarlark(iter.Key().Interface())
+				if err != nil {
+					return nil, err
+				}
 
-			if err = dict.SetKey(key, val); err != nil {
-				return nil, err
+				val, err := convertToStarlark(iter.Value().Interface())
+				if err != nil {
+					return nil, err
+				}
+				if err = dict.SetKey(key, val); err != nil {
+					return nil, err
+				}
 			}
+			return dict, nil
 		}
-		return dict, nil
 	}
 	return nil, nil
```

Comment:
Return an error if the value can't be converted? Additionally, I think we need to avoid leaking a golang `nil` into a `starlark.Value`. 

> Starlark's None value is not equal to Go's nil, but nil may be assigned to a Starlark Value. Be careful to avoid allowing Go nil values to leak into Starlark data structures.

https://godoc.org/go.starlark.net/starlark

Reply:
Yeah I will return an error. Unsure where I stand. I like seeing the <nil> in the output. If it fails to convert we have big problems. Most of these are fine now unless starlark have complex items likely.